### ISSUE TITLE
batch size effect on YOLO26 reproduction

### ISSUE THREAD

**Issue by mountain111** [ultralytics/ultralytics#24287]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

environment: torch2.5.1+cu11.8

hyper-parameters are setup as those reported officially.   

COCO2017 detection, mAP50-95 are compared as the following table

<img width="400" height="150" alt="Image" src="https://github.com/user-attachments/assets/9656f570-962c-43d9-bcb8-15f6add8443e" />

My questions:

(1) The results increase much slowly with batch=256 than those with bach=128. Considering nominal batch size rule，why are the differences so large?

(2) The results reproduced  with batch=128 are also lower than the reported those.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @mountain111, thank you for your interest in Ultralytics 🚀! This is an automated response to help get your issue moving quickly, and an Ultralytics engineer will also assist you soon.
We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.
Since this appears to be a training/reproducibility ❓ question, please provide as much information as possible, including:
- the exact training command or script used
- the full model variant and config
- the complete training logs
- your hardware details and number of GPUs
- whether AMP, deterministic settings, cache, workers, and data augmentation settings were changed
- the exact dataset preparation and evaluation steps
- results for multiple runs if available
Please also verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).
If you believe this may instead be a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.
Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.
## Upgrade
Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:
```bash
pip install -U ultralytics
```
## Environments
YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):
- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>
## Status
<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
> The results increase much slowly with batch=256 than those with bach=128. Considering nominal batch size rule，why are the differences so large?

There are fewerr iterations. If you're not compensating with increase in LR, then that's expected.

> The results reproduced with batch=128 are also lower than the reported those.

Where is this official results from?

**Comment by Y-T-G** [comment]
By default, nominal batch size is 64. So the number of optimizer steps would stay same for batches 2, 4, 8, 16, 32, 64. But if you go beyond 64, you're over the nominal batch size, and you have lower steps.

Also even with gradient accumulation, the batch size still has an impact because batch norm can't be accumulated.

**Comment by mountain111** [comment]
> > The results increase much slowly with batch=256 than those with bach=128. Considering nominal batch size rule，why are the differences so large?
> 
> There are fewerr iterations. If you're not compensating with increase in LR, then that's expected.
> 
> > The results reproduced with batch=128 are also lower than the reported those.
> 
> Where is this official results from?

Does LR need to be increased manually in config file?

The official results are obtained via the released model and model.ckpt['train_results'].

**Comment by Y-T-G** [comment]
The official traint uses Objects365 pretrained weights.

And the dataset needs to be `coco.yaml` which is segmentation dataset, not detection. Because segmentation labels are more precise and used to generate the detection boxes automatically after augmentations

**Comment by Y-T-G** [comment]
> Does LR need to be increased manually in config file?

Yes
